### ISSUE TITLE
package option for elmer extensions

### ISSUE THREAD

**Issue by simonpcouch** [tidyverse/elmer#175]
Been working on a couple BYO-key elmer extensions that aim to support any of the `chat_*()` backends from elmer, like [this one](https://simonpcouch.github.io/pal/reference/dot-init_pal.html) or [this one](https://github.com/simonpcouch/testhelper/blob/e00a68ae309e40dc8682e12b29148b56547ec594/R/test_helper.R#L41-L44). For each of them, I assume that folks are likely to use the same provider and model most of the time, so I support toggling those arguments using package-level options and recommend setting them in `.Rprofile`, like:

```r
options(
  .pal_fn = "chat_openai",
  .pal_args = list(model = "gpt-4o-mini")
)
```

or:

```r
options(
  .testhelper_fn = "chat_openai",
  .testhelper_args = list(model = "gpt-4o-mini")
)
```

While I can see the value in having different tools hook into different models by default, I think it'd also be nice to have more general `.elmer_fn` or `.elmer_args` options that would apply across extensions. Would yall consider documenting some "recommended" interface for extension options that include more general options like those?

**Comment by hadley** [comment]
Maybe we could have a `chat_default()` that used a user-supplied default? (Or maybe picked automatically based on presence of env vars?)

@jcheng5 what do you think?

**Comment by hadley** [comment]
@simonpcouch I think I'll probably export the provider classes, and then some generic way to create a chat object from a provider. Then we could possibly have `ellmer::set_default_provider()` which packages could use. I do wonder a little bit about making this too easy since this does effectively give packages the ability to spend money on your behalf, so we might need to think about some explicit acknowledgment workflow.

**Comment by simonpcouch** [comment]
Ooo I do like this interface. The hesitancy re: spending money without user consent definitely makes sense and I'm happy to work with that acknowledgement workflow; even if a user of pal/ensure/gander has to respond to a y/n prompt, this is an easier setup workflow than setting those package options in `.Rprofile`.

**Comment by hadley** [comment]
BTW I realised that probably this should be a single option that creates a chat object:

```R
options(
  pal_chat = function(...) {
    ellmer::chat_openai(..., model = "gpt-4o-mini")
  }
)
```

**Comment by simonpcouch** [comment]
Gave this a go for pal and gander and I generally do prefer this over the pair of arguments. I don't actually need to pass anything through `...` in either package, though, so I'm a bit conflicted on whether this option ought to just be the Chat itself:

```r
options(pal_chat = ellmer::chat_openai(..., model = "gpt-4o-mini"))
```

...instead of a function that returns the Chat:

```r
options(pal_chat = function() ellmer::chat_openai(..., model = "gpt-4o-mini"))
```

I can imagine some side effects of executing e.g. `ellmer::chat_openai(..., model = "gpt-4o-mini")` on startup, but it'd be nice that 1) the option is indeed a "chat" as suggested by the option name and 2) the user doesn't need to add nor understand the `function() ` boilerplate.

Feel like I might be missing something here, so I implemented it as you suggest in my first go. Neither of those PRs are merged yet, though—certainly curious to hear your thoughts.

**Comment by hadley** [comment]
Yeah if you only need an existing chat object the first should be fine. You’ll just need to make sure to clone it so you don’t end up with all the history in one chat.

**Comment by hadley** [comment]
I think for now, we'll handle this via #361 and then see how that goes.
